src/components/organisms/SchedulePageModal/SchedulePageModal.tsx

```diff
@@ -1,162 +1,126 @@
+
+  const { relatedVenueEvents, relatedVenues } = useConnectRelatedVenues({
     venueId,
     withEvents: true,
   });
 
-  const relatedVenuesById: Partial<
-    Record<string, WithId<AnyVenue>>
-  > = relatedVenues.reduce(itemsToObjectByIdReducer, {});
+  const [relatedRooms, setRelatedRooms] = useState<Room[]>([]);
+
+  useEffect(() => {
+    const rooms: Room[] = [];
+    relatedVenues
+      .map((venue) => venue.rooms || [])
+      .forEach((venueRooms) => rooms.push(...venueRooms));
+    setRelatedRooms(rooms);
+  }, [relatedVenues]);
 
-  const orderedEvents: DatedEvents = useMemo(() => {
+  const schedule: ScheduleDay[] = useMemo(() => {
     const liveAndFutureEvents = relatedVenueEvents.filter(isEventLiveOrFuture);
-    const hasEvents = liveAndFutureEvents.length > 0;
-
-    const nowDay = startOfDay(new Date());
-
-    const dates: DatedEvents = range(0, DAYS_AHEAD).map((idx) => {
-      const day = addDays(nowDay, idx);
-
-      const todaysEvents = liveAndFutureEvents
-        .filter((event) => {
-          return isWithinInterval(day, {
-            start: startOfDay(new Date(event.start_utc_seconds * 1000)),
-            end: endOfDay(
-              new Date(
-                (event.start_utc_seconds + event.duration_minutes * 60) * 1000
-              )
-            ),
-          });
-        })
-        .sort((a, b) => a.start_utc_seconds - b.start_utc_seconds);
-
-      return {
-        dateDay: day,
-        events: hasEvents ? todaysEvents : [],
-      };
-    });
-
-    return dates;
-  }, [relatedVenueEvents]);
-
-  const [date, setDate] = useState(0);
-
-  const scheduleTabs = useMemo(
+    const today = startOfDay(Date.now());
+    const buildScheduleEvent = scheduleDayBuilder(
+      today,
+      liveAndFutureEvents,
+      relatedRooms,
+      userEventIds
+    );
+
+    return range(0, DAYS_AHEAD).map((dayIndex) => buildScheduleEvent(dayIndex));
+  }, [relatedVenueEvents, relatedRooms, userEventIds]);
+
+  const [selectedDayIndex, setSelectedDayIndex] = useState(0);
+
+  const onWeekdayClick = useCallback(
+    (index: number): MouseEventHandler<HTMLLIElement> => (e) => {
+      e.stopPropagation();
+      setSelectedDayIndex(index);
+    },
+    []
+  );
+
+  const weekdayClasses = useCallback(
+    (index) =>
+      classNames("SchedulePageModal__weekday", {
+        "SchedulePageModal__weekday--active": index === selectedDayIndex,
+      }),
+    [selectedDayIndex]
+  );
+
+  const weekdays = useMemo(
     () =>
-      orderedEvents.map((day, idx) => (
+      schedule.map((day, index) => (
         <li
-          key={formatDate(day.dateDay.getTime())}
-          className={`button ${idx === date ? "active" : ""}`}
-          onClick={(e) => {
-            e.stopPropagation();
-            setDate(idx);
-          }}
+          key={formatDate(day.dayStartUtcSeconds)}
+          className={weekdayClasses(index)}
+          onClick={onWeekdayClick(index)}
```

Comment:
Instead of using the position `index` here, can we use something from `day` to store our 'selected tab' state? Maybe the name of the day/similar?

Reply:
Actually, looking at `Schedule[]` that might be more effort than it's worth.

Reply:
hmm, was thinking about that. If we capture date instead of index it can behave wrong when current date is changed after midnight.
But if we have index saved, we'll always have "Today" selected.
what do you say?

Reply:
We'll have 'today' selected still as you said, but will all of the other bits of code that use the `index` still work correctly?

Regardless, I think we probably leave it as `index` (but worth thinking through if there will be any other bugs based on that at midnight/etc)